sys/linux/dev_vfio.txt

```diff
@@ -0,0 +1,51 @@
+# Copyright 2017 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+# https://www.kernel.org/doc/Documentation/vfio.txt
+# https://elixir.bootlin.com/linux/latest/source/drivers/vfio/vfio.c
+# https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/vfio.h
+# https://elixir.bootlin.com/linux/latest/source/drivers/vfio/vfio_iommu_type1.c
+
+include <uapi/linux/fcntl.h>
+include <linux/vfio.h>
+
+define VFIO_IOMMU_INFO_PGSIZES	(1 << 0)
+define VFIO_DMA_MAP_FLAG_READ	(1 << 0)
+define VFIO_DMA_MAP_FLAG_WRITE	(1 << 1)
+
+resource fd_vfio[fd]
+
+openat$vfio(fd const[AT_FDCWD], file ptr[in, string["/dev/vfio/vfio"]], flags flags[open_flags], mode const[0]) fd_vfio
+
+ioctl$VFIO_GET_API_VERSION(fd fd_vfio, cmd const[VFIO_GET_API_VERSION])
+ioctl$VFIO_CHECK_EXTENSION(fd fd_vfio, cmd const[VFIO_CHECK_EXTENSION], arg intptr)
```

Comment:
This is not clear from the kernel source code, but `arg` is supposed to be an identifier of one the extensions, which are defined [here](https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/vfio.h#L23). One way to find this out is to google for `VFIO_CHECK_EXTENSION` and find e.g. [this](https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/vfio.h#L23). Even though it doesn't really matter here, but it's better to use `flags` type and pass in those extention codes.